src/domain/use_cases/experiment.py

```
@@ -72,69 +102,92 @@ def compute_variability(self, dataset: Dataset) -> float:
+                result = ExperimentResult(
+                    dataset=dataset_name,
+                    model_weights=weights_path,
+                    variability=variability_score,
+                    similarity=similarity_score,
+                    num_samples=len(dataset),
+                    num_generated=self.num_generated_images,
+                )
+                results.append(result)
 
             except Exception as error:
                 logger.error(f"Failed to process {dataset_name}: {error}")
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)